File: QytetetJava/src/modeloqytetet/Calle.java
/**
 * Practica de:
 * 
 * Francisco Beltrán Sánchez
 * Pablo Cordero Romero
 */
package modeloqytetet;

public class Calle extends Casilla{
    private TituloPropiedad titulo;

    Calle(int numeroCasilla, int coste, TituloPropiedad ttl){
        super(numeroCasilla, coste);
        this.titulo = ttl;
    }

    @Override
    protected TipoCasilla getTipo(){
        return TipoCasilla.CALLE;
    }

    @Override
    protected boolean soyEdificable(){
        return true;
    }

    @Override
    protected TituloPropiedad getTitulo() {
        return titulo;
    }

    @Override
    public boolean tengoPropietario(){
        return this.titulo.tengoPropietario();
    }

    public TituloPropiedad asignarPropietario(Jugador jugador){
        titulo.setPropietario(jugador);

        return titulo;
    }

    private void setTitulo(TituloPropiedad tp){
        this.titulo = tp;
        super.setCoste(this.titulo.getPrecioCompra());
    }

    public int pagarAlquiler(){
        return titulo.pagarAlquiler();
    }

    public boolean propietarioEncarcelado(){
        return this.titulo.propietarioEncarcelado();
    }

    @Override
    public String toString() {
        return super.toString() + "\n\ttipo= CALLE" + "\n\ttitulo=" + titulo;
    }
    
}

File: QytetetRuby/lib/calle.rb
#encoding: utf-8
# Francisco Beltrán Sánchez
# Pablo Cordero Romero
module ModeloQytetet
    class Calle < Casilla
        attr_reader :titulo
        attr_reader :tipo

        def initialize(numCas, coste, ttl)
            super(numCas, coste)
            @titulo = ttl
            @tipo = TipoCasilla::CALLE
        end

#        def self.copia(casilla,ttl)
#            super(casilla)
#            @titulo = ttl
#            @tipo = TipoCasilla::CALLE
#            
#            self
#        end

        def soyEdificable
            salida = true
            salida
        end

        def tengoPropietario
            salida = @titulo.tengoPropietario();
            salida
        end

        def setTitulo(ttl)
            @titulo = ttl
            super(ttl)
        end

        def propietarioEncarcelado
            @titulo.propietarioEncarcelado
        end

        def asignarPropietario(jugador)
            @titulo.propietario = jugador
            @titulo
        end

        def pagarAlquiler
            @titulo.pagarAlquiler
        end

        def to_s
          "#{super}\n\tTipo: #{@tipo} \n\tTitulo: #{@titulo} \n"
        end

        private :setTitulo
    end
end